source/extensions/filters/http/proto_api_scrubber/scrubbing_util_lib/field_checker.h

```diff
@@ -32,10 +32,28 @@ using proto_processing_lib::proto_scrubber::ScrubberContext;
 class FieldChecker : public FieldCheckerInterface, public Logger::Loggable<Logger::Id::filter> {
 public:
   FieldChecker(const ScrubberContext scrubber_context,
-               const Envoy::StreamInfo::StreamInfo* stream_info, const std::string& method_name,
+               const Envoy::StreamInfo::StreamInfo* stream_info,
+               const Http::RequestHeaderMap* request_headers,
+               const Http::ResponseHeaderMap* response_headers,
+               const Http::RequestTrailerMap* request_trailers,
+               const Http::ResponseTrailerMap* response_trailers, const std::string& method_name,
                const ProtoApiScrubberFilterConfig* filter_config)
       : scrubber_context_(scrubber_context), matching_data_(*stream_info),
-        method_name_(method_name), filter_config_ptr_(filter_config) {}
+        method_name_(method_name), filter_config_ptr_(filter_config) {
+    // Populate matching data with whatever is available.
```

Comment:
FWIW, the design here seems a bit leaky (a single field checker for both request and response, but used differently for each). I guess that is ok for now, but if further divergence from behavior needs to be done for request/response, it may be best to divide the two paths to different classes.